core/src/main/java/cz/xtf/core/waiting/SupplierWaiter.java

```diff
@@ -86,6 +99,27 @@ public SupplierWaiter logPoint(LogPoint logPoint) {
 		return this;
 	}
 
+	public SupplierWaiter onSuccess(Runnable runnable) {
+		onSuccess = runnable;
+		return this;
+	}
+
+	public SupplierWaiter onFailure(Runnable runnable) {
+		onFailure = runnable;
+		return this;
+	}
+
+	public SupplierWaiter onTimeout(Runnable runnable) {
+		onTimeout = runnable;
+		return this;
+	}
+
+	public SupplierWaiter reactOn(Function<X, Boolean> function, Runnable runnable) {
```

Comment:
If we are not exposing the reactOn in the interface anyway (and the use case is really only for the implementation of a specific waiter, I think we can solve this with inheritance instead?

(e.g. have 

```
protected onIteration() {
  // empty
}
```

called inside waitFor on each iteration, that the speciifc waiter could override?

Reply:
(and  if you insist on having this dynamically configirable, I'd prefer to have onIteration(Runnable) ... having a specific "add if (condifition) then (runnable)" looks oddly specific.